pricegraph/src/lib.rs

```diff
@@ -84,6 +84,37 @@ impl TransitiveOrder {
     }
 }
 
+/// A struct representing a market.
+///
+/// This is used for computing transitive orderbooks.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub struct Market {
+    /// The base token being purchased in a market.
+    pub base: TokenId,
+    /// The quote token being sold in a market. Prices in a market are always
+    /// expressed in the quote token.
+    pub quote: TokenId,
+}
+
+impl Market {
+    /// Returns the token pair for ask orders.
+    pub fn ask_pair(self) -> TokenPair {
```

Comment:
It's a bit counter-intuitive to me that you can turn a TokenPair into one market but one market into two token pairs. Maybe we should clarify the `Into` more explicitly into something like from_buy_order?

Reply:
Oh! You mean the `price_estimator::models::TokenPair`. I'll rename that, I think that type should be a "market" anyway.

Reply:
@fleupold  Hey, I refactored the `price_estimator::models::TokenPair` to not use `Into<pricegraph::TokenPair>` and `Into<pricegraph::Market>` which was admittedly a bit confusing. Its now `price_estimator::models::Market` and implements `Deref<Target = Market>` and we have a `Filter` prefix that gets the bid token pair for a market. Let me know if this is a little clearer.